parlai/core/torch_ranker_agent.py

```diff
@@ -586,31 +606,33 @@ def _build_candidates(self, batch, source, mode):
                         )
 
         elif source == 'fixed':
-            warn_once(
-                "[ Executing {} mode with a common set of fixed candidates "
-                "(n = {}). ]".format(mode, len(self.fixed_candidates))
-            )
             if self.fixed_candidates is None:
                 raise ValueError(
                     "If using candidate source 'fixed', then you must provide the path "
-                    "to a file of candidates with the flag --fixed-candidates-path"
+                    "to a file of candidates with the flag --fixed-candidates-path or "
+                    "the name of a task with --fixed-candidates-task."
                 )
+            warn_once(
+                "[ Executing {} mode with a common set of fixed candidates "
+                "(n = {}). ]".format(mode, len(self.fixed_candidates))
+            )
 
             cands = self.fixed_candidates
             cand_vecs = self.fixed_candidate_vecs
 
             if label_vecs is not None:
                 label_inds = label_vecs.new_empty((batchsize))
                 bad_batch = False
-                for i, label_vec in enumerate(label_vecs):
-                    label_vec_pad = label_vec.new_zeros(cand_vecs[i].size(0)).fill_(
```

Comment:
cand_vecs here are fixed and not built in a batch but the `i` here was trying to get "batch row i" from the candidate vectors

Reply:
the other changes are primarily just renaming to variables that are more clear (better for us to use "batch_idx" rather than "i")